lib/spack/spack/database.py

```diff
@@ -187,7 +187,8 @@ def __init__(
             ref_count=0,
             explicit=False,
             installation_time=None,
-            deprecated_for=None
+            deprecated_for=None,
+            cache_exists=False,
```

Comment:
Both of these needs docs in the docstring.

Can we change `cache_exists`  to `in_buildcache` -- I think `cache_exists` could be taken as "exists somewhere", but `in_buildcache` makes it more clear it's only for the buildcache described by the DB.

A docstring would probably help with that as well, so maybe that is all we need, but something needs to describe what this thing means so that we'll remember later.